utils/config_file_impl.hh

```diff
@@ -202,13 +203,21 @@ void utils::config_file::named_value<T>::set_value(const YAML::Node& node) {
     _source = config_source::SettingsFile;
 }
 
+template<>
+bool utils::config_file::lexical_cast<bool>(sstring value);
```

Comment:
Nit: doesn't need to be a template, overload resolution will pick a non-template over a template.

Reply:
@avikivity but the function signature has a different return type and same parameter type.
I don't quite follow what you have in mind.

Reply:
Apparently it does need to be a template, since `utils::config_file::named_value<T>::set_value` calls it like this: `(*this)(lexical_cast<T>(value), src);`